src/plugins/intel_cpu/src/nodes/rope.cpp

```diff
@@ -40,12 +307,34 @@ struct RoPE::RoPEExecutorRotateHalf : public RoPE::Executor {
         ov::intel_cpu::PlainTensor t_sin(inputs[2]);
         ov::intel_cpu::PlainTensor t_dst(outputs[0]);
         ov::intel_cpu::PlainTensor gather;
+        auto rotary_dims = config.rotary_ndims;
 
+        if (!m_init) {
+            m_init = true;
+#ifdef OPENVINO_ARCH_X86_64
+            jit_rotary_compile_params jcp;
+            jcp.src_prc = precision_of<T>::value;
+            jcp.dst_prc = precision_of<T>::value;
+            jcp.rotary_ndims = rotary_dims;
+            jcp.interleave = false;
+            if (mayiuse(cpu_isa_t::avx512_core)) {
+                m_rotaryKernel.reset(new jit_rotary_kernel<cpu_isa_t::avx512_core>(jcp));
+            } else if (mayiuse(cpu_isa_t::avx2)) {
+                m_rotaryKernel.reset(new jit_rotary_kernel<cpu_isa_t::avx2>(jcp));
+            }
+            if (m_rotaryKernel)
+                m_rotaryKernel->create_ker();
+#endif
+        }
```

Comment:
Logically it should be part of Executor ctor, then m_init field will not be needed

Reply:
Moved into ctor.